msrestazure/azure_exceptions.py

```diff
@@ -261,6 +274,6 @@ def __init__(self, type, info):
 
     def __str__(self):
         """Cloud error message."""
-        error_str = "Type: {}".format(self.type)
-        error_str += "\nInfo: {}".format(json.dumps(self.info, indent=4))
+        error_str = u"Type: {}".format(self.type)
+        error_str += u"\nInfo: {}".format(json.dumps(self.info, indent=4))
```

Comment:
What happens here is self.info contains non-unicode values? Shouldn't we also pass `ensure_ascii = False` to fully handle that?

Reply:
Good point
```python
>>> json.dumps({u'foo': u'bar'})
'{"foo": "bar"}'
>>> json.dumps({u'foo': u'bar'}, ensure_ascii=False)
u'{"foo": "bar"}'
```

Probably worth it here

Reply:
Though it probably won't change a lot for a `str` usage here:
```python
>>> json.dumps({u'foo': u'ééé'}, ensure_ascii=False)
u'{"foo": "\xe9\xe9\xe9"}'
>>> json.dumps({u'foo': u'ééé'})
'{"foo": "\\u00e9\\u00e9\\u00e9"}'
```

Both are technically correct dumps of JSON

Reply:
Actually on a print the first one looks better:
```python
>>> print(json.dumps({u'foo': u'ééé'}, ensure_ascii=False))
{"foo": "ééé"}
```

So I will add the ensure_ascii, thanks @johanste didn't realized that